File: Algorithms/Java/Best Time to Buy and Sell Stock.java
public class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
		if (prices.length < 2)
			return profit;
		int maximum = prices[0], minimum = prices[0];
		for (int x : prices) {
			if (x < minimum) {
				minimum = x;
				maximum = x;
			} else if (x >= maximum) {
				maximum = x;
				profit = (maximum - minimum) > profit ? maximum - minimum
						: profit;
			}
		}
		return profit;
    }
}

File: Algorithms/Python/Best-Time-to-Buy-and-Sell-Stock.py
class Solution(object):

    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0
        ans, minNum = 0, prices[0]
        for i in range(1, len(prices)):
            if prices[i] > minNum:
                ans = max(prices[i] - minNum, ans)
            else:
                minNum = prices[i]
        return ans
